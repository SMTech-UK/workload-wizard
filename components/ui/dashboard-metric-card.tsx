import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-500">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-muted-foreground mt-1">{subtitle}</div>}
    </div>
  );
} 