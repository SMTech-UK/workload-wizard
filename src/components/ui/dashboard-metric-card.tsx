import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white dark:bg-zinc-900 shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-900 dark:text-white">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-gray-600 dark:text-gray-300 mt-1">{subtitle}</div>}
    </div>
  );
} 