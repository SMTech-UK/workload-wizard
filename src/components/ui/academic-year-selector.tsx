"use client"

import React from "react";
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
import { Badge } from "@/components/ui/badge";
import { useAcademicYear } from "@/hooks/useAcademicYear";
import { Calendar } from "lucide-react";

interface AcademicYearSelectorProps {
  className?: string;
  showActiveBadge?: boolean;
  onAcademicYearChange?: (academicYearId: string) => void;
}

export function AcademicYearSelector({ 
  className, 
  showActiveBadge = true,
  onAcademicYearChange 
}: AcademicYearSelectorProps) {
  const { 
    academicYears, 
    currentAcademicYear, 
    currentAcademicYearId, 
    changeAcademicYear 
  } = useAcademicYear();

  const handleAcademicYearChange = async (academicYearId: string) => {
    try {
      await changeAcademicYear(academicYearId);
      onAcademicYearChange?.(academicYearId);
    } catch (error) {
      console.error('Failed to change academic year:', error);
    }
  };

  // Function to format academic year name (e.g., "2025/26" -> "AY-25/26")
  const formatAcademicYearName = (yearName: string) => {
    // Extract the last 2 digits from each year part
    const parts = yearName.split('/');
    if (parts.length === 2) {
      const startYear = parts[0].slice(-2);
      const endYear = parts[1].slice(-2);
      return `AY-${startYear}/${endYear}`;
    }
    // Fallback to original name if format is unexpected
    return yearName;
  };

  if (!academicYears || academicYears.length === 0) {
    return (
      <div className={`flex items-center gap-2 text-muted-foreground ${className}`}>
        <Calendar className="h-4 w-4" />
        <span className="text-sm">No academic years configured</span>
      </div>
    );
  }

  return (
    <div className={`flex items-center gap-2 ${className}`}>
      <Select
        value={currentAcademicYearId || ""}
        onValueChange={handleAcademicYearChange}
      >
        <SelectTrigger className="w-auto min-w-[120px] flex items-center gap-2">
          <Calendar className="h-4 w-4 text-muted-foreground" />
          <span className="font-medium">
            {currentAcademicYear ? formatAcademicYearName(currentAcademicYear.name) : "Select AY"}
          </span>
        </SelectTrigger>
        <SelectContent>
          {academicYears.map((year: any) => (
            <SelectItem key={year._id} value={year._id}>
              <div className="flex items-center gap-2">
                <Calendar className="h-4 w-4 text-muted-foreground" />
                <span>{formatAcademicYearName(year.name)}</span>
                {year.isActive && showActiveBadge && (
                  <Badge variant="default" className="text-xs">Active</Badge>
                )}
                {year.isStaging && (
                  <Badge variant="secondary" className="text-xs">Staging</Badge>
                )}
              </div>
            </SelectItem>
          ))}
        </SelectContent>
      </Select>
    </div>
  );
} 