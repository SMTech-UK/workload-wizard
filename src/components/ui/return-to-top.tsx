"use client";

import { Button } from "@/components/ui/button";
import { ChevronUp, Home } from "lucide-react";
import { useState, useEffect } from "react";
import Link from "next/link";

export function ReturnToTop() {
  const [isVisible, setIsVisible] = useState(true); // Temporarily always visible for testing

  useEffect(() => {
    const toggleVisibility = () => {
      console.log("Scroll position:", window.pageYOffset);
      if (window.pageYOffset > 300) {
        console.log("Setting visible to true");
        setIsVisible(true);
      } else {
        console.log("Setting visible to false");
        setIsVisible(false);
      }
    };

    // Call once on mount to check initial state
    toggleVisibility();
    
    window.addEventListener("scroll", toggleVisibility);
    return () => window.removeEventListener("scroll", toggleVisibility);
  }, []);

  const scrollToTop = () => {
    console.log("Scrolling to top");
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  console.log("ReturnToTop component render, isVisible:", isVisible);

  return (
    <>
      {/* Return to Home button - top left */}
      <Link href="/">
        <Button
          size="sm"
          variant="outline"
          className="fixed top-4 left-4 z-50 shadow-lg hover:shadow-xl transition-all duration-200"
          aria-label="Return to home"
        >
          <Home className="h-4 w-4 mr-2" />
          Home
        </Button>
      </Link>

      {/* Return to Top button - bottom left */}
      <Button
        onClick={scrollToTop}
        size="sm"
        className="fixed bottom-4 left-4 z-50 rounded-full shadow-lg hover:shadow-xl transition-all duration-200 bg-blue-500 text-white"
        aria-label="Return to top"
      >
        <ChevronUp className="h-4 w-4" />
      </Button>
    </>
  );
} 