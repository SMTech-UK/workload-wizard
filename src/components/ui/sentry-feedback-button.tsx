"use client";

import { useEffect, useState } from "react";
import * as Sentry from "@sentry/nextjs";
import { Button } from "@/components/ui/button";
import { Bug, MessageCircle } from "lucide-react";
import { useModalAwareButton } from "@/hooks/useModalAwareButton";
import { createPortal } from "react-dom";

export function SentryFeedbackButton() {
  const [feedback, setFeedback] = useState<any>(null);
  const [widget, setWidget] = useState<any>(null);
  const [mounted, setMounted] = useState(false);
  const buttonRef = useModalAwareButton();

  useEffect(() => {
    setMounted(true);
    // Get the feedback instance on the client side
    setFeedback(Sentry.getFeedback());
  }, []);

  const handleFeedbackClick = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    
    if (!feedback) return;

    if (widget) {
      // If widget is open, close it
      widget.removeFromDom();
      setWidget(null);
    } else {
      // Create and show the form immediately
      feedback.createForm().then((newWidget: any) => {
        newWidget.appendToDom();
        newWidget.open();
        setWidget(newWidget);
      }).catch((error: any) => {
        console.error('Failed to create Sentry feedback form:', error);
      });
    }
  };

  if (!feedback || !mounted) return null;

  const button = (
    <Button
      ref={buttonRef}
      variant="destructive"
      size="lg"
      onClick={handleFeedbackClick}
      onMouseDown={(e) => {
        e.preventDefault();
        e.stopPropagation();
      }}
      className="fixed bottom-4 right-4 z-[99999] shadow-lg"
      style={{
        zIndex: 99999,
        pointerEvents: 'auto',
      }}
      aria-label="Report a bug or give feedback"
    >
      <Bug className="h-4 w-4 mr-2" />
      Report an Issue
    </Button>
  );

  // Render the button in a portal to ensure it's outside modal context
  return createPortal(button, document.body);
} 