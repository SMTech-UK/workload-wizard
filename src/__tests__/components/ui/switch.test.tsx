import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Switch } from '@/components/ui/switch';

// Custom query to find elements by data-slot attribute
const getByDataSlot = (container: HTMLElement, slotName: string) => {
  return container.querySelector(`[data-slot="${slotName}"]`);
};

const getAllByDataSlot = (container: HTMLElement, slotName: string) => {
  return container.querySelectorAll(`[data-slot="${slotName}"]`);
};

describe('Switch Component - Comprehensive UI Testing', () => {
  describe('Basic Rendering and Functionality', () => {
    it('should render switch with default styling', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toBeInTheDocument();
      expect(switchElement).toHaveClass(
        'peer', 'inline-flex', 'h-[1.15rem]', 'w-8', 'shrink-0', 'items-center', 'rounded-full', 'border', 'border-transparent', 'shadow-xs'
      );
    });

    it('should render switch with custom className', () => {
      // Arrange
      const { container } = render(<Switch className="custom-switch" />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('custom-switch');
      expect(switchElement).toHaveClass('peer', 'inline-flex', 'h-[1.15rem]', 'w-8', 'shrink-0', 'items-center', 'rounded-full');
    });

    it('should render switch thumb', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const thumb = getByDataSlot(container, 'switch-thumb');
      expect(thumb).toBeInTheDocument();
      expect(thumb).toHaveClass(
        'bg-background', 'pointer-events-none', 'block', 'size-4', 'rounded-full', 'ring-0', 'transition-transform'
      );
    });

    it('should render switch with default unchecked state', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      expect(switchElement).not.toBeChecked();
      expect(switchElement).toHaveAttribute('data-state', 'unchecked');
    });

    it('should render switch with checked state', () => {
      // Arrange
      const { container } = render(<Switch defaultChecked />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      expect(switchElement).toBeChecked();
      expect(switchElement).toHaveAttribute('data-state', 'checked');
    });
  });

  describe('Accessibility Testing', () => {
    it('should have proper semantic structure', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement?.tagName).toBe('BUTTON');
      expect(switchElement).toHaveAttribute('data-slot', 'switch');
      expect(switchElement).toHaveAttribute('type', 'button');
    });

    it('should support ARIA attributes', () => {
      // Arrange
      const { container } = render(<Switch aria-label="Toggle notifications" />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('aria-label', 'Toggle notifications');
    });

    it('should support aria-checked attribute', () => {
      // Arrange
      const { container } = render(<Switch defaultChecked />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('aria-checked', 'true');
    });

    it('should support role attribute', () => {
      // Arrange
      const { container } = render(<Switch role="switch" />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('role', 'switch');
    });

    it('should be keyboard accessible', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('tabIndex', '0');
    });

    it('should support focus-visible styling', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('focus-visible:border-ring', 'focus-visible:ring-ring/50', 'focus-visible:ring-[3px]');
    });
  });

  describe('User Interaction Testing', () => {
    it('should toggle state when clicked', async () => {
      // Arrange
      const user = userEvent.setup();
      const { container } = render(<Switch />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      await user.click(switchElement);

      // Assert
      expect(switchElement).toBeChecked();
      expect(switchElement).toHaveAttribute('data-state', 'checked');
    });

    it('should toggle state when space key is pressed', async () => {
      // Arrange
      const user = userEvent.setup();
      const { container } = render(<Switch />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      switchElement.focus();
      await user.keyboard(' ');

      // Assert
      expect(switchElement).toBeChecked();
      expect(switchElement).toHaveAttribute('data-state', 'checked');
    });

    it('should toggle state when enter key is pressed', async () => {
      // Arrange
      const user = userEvent.setup();
      const { container } = render(<Switch />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      switchElement.focus();
      await user.keyboard('{Enter}');

      // Assert
      expect(switchElement).toBeChecked();
      expect(switchElement).toHaveAttribute('data-state', 'checked');
    });

    it('should call onChange handler when toggled', async () => {
      // Arrange
      const user = userEvent.setup();
      const handleChange = jest.fn();
      const { container } = render(<Switch onCheckedChange={handleChange} />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      await user.click(switchElement);

      // Assert
      expect(handleChange).toHaveBeenCalledWith(true);
    });

    it('should not toggle when disabled', async () => {
      // Arrange
      const user = userEvent.setup();
      const handleChange = jest.fn();
      const { container } = render(<Switch disabled onCheckedChange={handleChange} />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      await user.click(switchElement);

      // Assert
      expect(switchElement).toBeDisabled();
      expect(switchElement).toHaveClass('disabled:cursor-not-allowed', 'disabled:opacity-50');
      expect(handleChange).not.toHaveBeenCalled();
    });

    it('should handle controlled component properly', async () => {
      // Arrange
      const user = userEvent.setup();
      const handleChange = jest.fn();
      const { container, rerender } = render(<Switch checked={false} onCheckedChange={handleChange} />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      await user.click(switchElement);

      // Assert
      expect(handleChange).toHaveBeenCalledWith(true);
      expect(switchElement).not.toBeChecked(); // Should remain false as it's controlled

      // Update to checked state
      rerender(<Switch checked={true} onCheckedChange={handleChange} />);
      expect(switchElement).toBeChecked();
    });
  });

  describe('Responsive Behavior', () => {
    it('should maintain proper sizing on mobile screens', () => {
      // Arrange
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 375,
      });

      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('h-[1.15rem]', 'w-8');
      expect(switchElement).toHaveClass('rounded-full');
    });

    it('should maintain proper sizing on tablet screens', () => {
      // Arrange
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 768,
      });

      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('h-[1.15rem]', 'w-8');
      expect(switchElement).toHaveClass('rounded-full');
    });

    it('should maintain proper sizing on desktop screens', () => {
      // Arrange
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 1024,
      });

      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('h-[1.15rem]', 'w-8');
      expect(switchElement).toHaveClass('rounded-full');
    });

    it('should handle custom sizing classes', () => {
      // Arrange
      const { container } = render(<Switch className="h-6 w-12" />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('h-6', 'w-12');
      expect(switchElement).toHaveClass('rounded-full');
    });
  });

  describe('State Management', () => {
    it('should handle checked state styling', () => {
      // Arrange
      const { container } = render(<Switch defaultChecked />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('data-[state=checked]:bg-primary');
    });

    it('should handle unchecked state styling', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('data-[state=unchecked]:bg-input');
    });

    it('should handle disabled state styling', () => {
      // Arrange
      const { container } = render(<Switch disabled />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('disabled:cursor-not-allowed', 'disabled:opacity-50');
      expect(switchElement).toBeDisabled();
    });

    it('should handle thumb positioning for checked state', () => {
      // Arrange
      const { container } = render(<Switch defaultChecked />);

      // Act & Assert
      const thumb = getByDataSlot(container, 'switch-thumb');
      expect(thumb).toHaveClass('data-[state=checked]:translate-x-[calc(100%-2px)]');
    });

    it('should handle thumb positioning for unchecked state', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      const thumb = getByDataSlot(container, 'switch-thumb');
      expect(thumb).toHaveClass('data-[state=unchecked]:translate-x-0');
    });
  });

  describe('Edge Cases and Error Handling', () => {
    it('should handle rapid clicking', async () => {
      // Arrange
      const user = userEvent.setup();
      const handleChange = jest.fn();
      const { container } = render(<Switch onCheckedChange={handleChange} />);

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      await user.click(switchElement);
      await user.click(switchElement);
      await user.click(switchElement);

      // Assert
      expect(handleChange).toHaveBeenCalledTimes(3);
      expect(handleChange).toHaveBeenNthCalledWith(1, true);
      expect(handleChange).toHaveBeenNthCalledWith(2, false);
      expect(handleChange).toHaveBeenNthCalledWith(3, true);
    });

    it('should handle multiple switches', async () => {
      // Arrange
      const user = userEvent.setup();
      const { container } = render(
        <div>
          <Switch data-testid="switch1" />
          <Switch data-testid="switch2" />
          <Switch data-testid="switch3" />
        </div>
      );

      // Act
      const switches = getAllByDataSlot(container, 'switch');
      await user.click(switches[0] as HTMLButtonElement);
      await user.click(switches[2] as HTMLButtonElement);

      // Assert
      expect(switches).toHaveLength(3);
      expect(switches[0]).toBeChecked();
      expect(switches[1]).not.toBeChecked();
      expect(switches[2]).toBeChecked();
    });

    it('should handle switch with form context', () => {
      // Arrange
      const { container } = render(
        <form>
          <Switch name="notifications" value="enabled" />
        </form>
      );

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('name', 'notifications');
      expect(switchElement).toHaveAttribute('value', 'enabled');
    });

    it('should handle switch with required attribute', () => {
      // Arrange
      const { container } = render(<Switch required />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('required');
    });

    it('should handle switch with aria-describedby', () => {
      // Arrange
      const { container } = render(<Switch aria-describedby="switch-description" />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveAttribute('aria-describedby', 'switch-description');
    });
  });

  describe('Snapshot Testing for UI Consistency', () => {
    it('should match snapshot for default switch', () => {
      // Arrange
      const { container } = render(<Switch />);

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for checked switch', () => {
      // Arrange
      const { container } = render(<Switch defaultChecked />);

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for disabled switch', () => {
      // Arrange
      const { container } = render(<Switch disabled />);

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for multiple switches', () => {
      // Arrange
      const { container } = render(
        <div>
          <Switch />
          <Switch defaultChecked />
          <Switch disabled />
        </div>
      );

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for switch with custom styling', () => {
      // Arrange
      const { container } = render(<Switch className="h-6 w-12" />);

      // Act & Assert
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Performance Testing', () => {
    it('should handle rapid re-renders efficiently', () => {
      // Arrange
      const { rerender } = render(<Switch />);

      // Act & Assert
      expect(screen.getByRole('button')).toBeInTheDocument();

      // Rerender with different props
      rerender(<Switch defaultChecked />);

      expect(screen.getByRole('button')).toBeChecked();
    });

    it('should handle className changes efficiently', () => {
      // Arrange
      const { rerender, container } = render(<Switch className="h-[1.15rem] w-8" />);

      // Act & Assert
      const switchElement = getByDataSlot(container, 'switch');
      expect(switchElement).toHaveClass('h-[1.15rem]', 'w-8');

      // Change className
      rerender(<Switch className="h-6 w-12" />);

      const updatedSwitch = getByDataSlot(container, 'switch');
      expect(updatedSwitch).toHaveClass('h-6', 'w-12');
      expect(updatedSwitch).not.toHaveClass('h-[1.15rem]', 'w-8');
    });

    it('should handle multiple switch renders efficiently', () => {
      // Arrange
      const switches = Array.from({ length: 100 }, (_, i) => `switch-${i + 1}`);

      const startTime = performance.now();

      // Act
      render(
        <div>
          {switches.map((id, index) => (
            <Switch key={index} data-testid={id} />
          ))}
        </div>
      );

      const endTime = performance.now();

      // Assert
      expect(screen.getByTestId('switch-1')).toBeInTheDocument();
      expect(screen.getByTestId('switch-100')).toBeInTheDocument();
      expect(endTime - startTime).toBeLessThan(1000); // Should render within 1 second
    });

    it('should handle state transitions efficiently', async () => {
      // Arrange
      const user = userEvent.setup();
      const { container } = render(<Switch />);

      const startTime = performance.now();

      // Act
      const switchElement = getByDataSlot(container, 'switch') as HTMLButtonElement;
      await user.click(switchElement);
      await user.click(switchElement);
      await user.click(switchElement);

      const endTime = performance.now();

      // Assert
      expect(switchElement).toBeChecked();
      expect(endTime - startTime).toBeLessThan(500); // Should transition within 500ms
    });
  });
}); 