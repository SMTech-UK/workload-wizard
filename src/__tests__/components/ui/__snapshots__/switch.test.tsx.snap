// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for checked switch 1`] = `
<button
  aria-checked="true"
  class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex h-[1.15rem] w-8 shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="switch"
  data-state="checked"
  role="switch"
  type="button"
  value="on"
>
  <span
    class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
    data-slot="switch-thumb"
    data-state="checked"
  />
</button>
`;

exports[`Switch Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for default switch 1`] = `
<button
  aria-checked="false"
  class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex h-[1.15rem] w-8 shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="switch"
  data-state="unchecked"
  role="switch"
  type="button"
  value="on"
>
  <span
    class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
    data-slot="switch-thumb"
    data-state="unchecked"
  />
</button>
`;

exports[`Switch Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for disabled switch 1`] = `
<button
  aria-checked="false"
  class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex h-[1.15rem] w-8 shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-disabled=""
  data-slot="switch"
  data-state="unchecked"
  disabled=""
  role="switch"
  type="button"
  value="on"
>
  <span
    class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
    data-disabled=""
    data-slot="switch-thumb"
    data-state="unchecked"
  />
</button>
`;

exports[`Switch Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for multiple switches 1`] = `
<div>
  <button
    aria-checked="false"
    class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex h-[1.15rem] w-8 shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
    data-slot="switch"
    data-state="unchecked"
    role="switch"
    type="button"
    value="on"
  >
    <span
      class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
      data-slot="switch-thumb"
      data-state="unchecked"
    />
  </button>
  <button
    aria-checked="true"
    class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex h-[1.15rem] w-8 shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
    data-slot="switch"
    data-state="checked"
    role="switch"
    type="button"
    value="on"
  >
    <span
      class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
      data-slot="switch-thumb"
      data-state="checked"
    />
  </button>
  <button
    aria-checked="false"
    class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex h-[1.15rem] w-8 shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
    data-disabled=""
    data-slot="switch"
    data-state="unchecked"
    disabled=""
    role="switch"
    type="button"
    value="on"
  >
    <span
      class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
      data-disabled=""
      data-slot="switch-thumb"
      data-state="unchecked"
    />
  </button>
</div>
`;

exports[`Switch Component - Comprehensive UI Testing Snapshot Testing for UI Consistency should match snapshot for switch with custom styling 1`] = `
<button
  aria-checked="false"
  class="peer data-[state=checked]:bg-primary data-[state=unchecked]:bg-input focus-visible:border-ring focus-visible:ring-ring/50 dark:data-[state=unchecked]:bg-input/80 inline-flex shrink-0 items-center rounded-full border border-transparent shadow-xs transition-all outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 h-6 w-12"
  data-slot="switch"
  data-state="unchecked"
  role="switch"
  type="button"
  value="on"
>
  <span
    class="bg-background dark:data-[state=unchecked]:bg-foreground dark:data-[state=checked]:bg-primary-foreground pointer-events-none block size-4 rounded-full ring-0 transition-transform data-[state=checked]:translate-x-[calc(100%-2px)] data-[state=unchecked]:translate-x-0"
    data-slot="switch-thumb"
    data-state="unchecked"
  />
</button>
`;
